src/otx/algo/instance_segmentation/maskrcnn.py

```diff
@@ -178,512 +123,198 @@ def _build_model(self, num_classes: int) -> TwoStageDetector:
+            # https://github.com/openvinotoolkit/training_extensions/pull/3431
+            loss_cls=CrossSigmoidFocalLoss(loss_weight=1.0, use_sigmoid=False),
+            loss_mask=CrossEntropyLoss(loss_weight=1.0, use_mask=True),
+            class_agnostic=False,
         )
 
         return TwoStageDetector(
             backbone=backbone,
             neck=neck,
             rpn_head=rpn_head,
             roi_head=roi_head,
-            train_cfg=train_cfg,
-            test_cfg=test_cfg,
+            roi_criterion=roi_criterion,
+            rpn_criterion=rpn_criterion,
         )
 
     @property
-    def _optimization_config(self) -> dict[str, Any]:
-        """PTQ config for MaskRCNN-SwinT."""
-        return {
-            "ignored_scope": {
-                "types": [
-                    "Add",
-                    "MVN",
-                    "Divide",
-                    "Multiply",
-                ],
-                "validate": False,
+    def _exporter(self) -> OTXModelExporter:
+        """Creates OTXModelExporter object that can export the model."""
+        if self.input_size is None:
+            msg = f"Input size attribute is not set for {self.__class__}"
+            raise ValueError(msg)
+
+        return OTXNativeModelExporter(
+            task_level_export_parameters=self._export_parameters,
+            input_size=(1, 3, *self.input_size),
+            mean=self.mean,
+            std=self.std,
+            resize_mode="fit_to_window",
+            pad_value=0,
+            swap_rgb=False,
+            via_onnx=True,
+            onnx_export_configuration={
+                "input_names": ["image"],
+                "output_names": ["boxes", "labels", "masks"],
+                "dynamic_axes": {
+                    "image": {0: "batch", 2: "height", 3: "width"},
+                    "boxes": {0: "batch", 1: "num_dets"},
+                    "labels": {0: "batch", 1: "num_dets"},
+                    "masks": {0: "batch", 1: "num_dets", 2: "height", 3: "width"},
+                },
+                "opset_version": 11,
+                "autograd_inlining": False,
             },
-            "preset": "mixed",
-        }
+            output_names=["bboxes", "labels", "masks"],
+        )
+
+    def load_from_otx_v1_ckpt(self, state_dict: dict, add_prefix: str = "model.") -> dict:
+        """Load the previous OTX ckpt according to OTX2.0."""
+        return OTXv1Helper.load_iseg_ckpt(state_dict, add_prefix)
+
+    @property
+    def _optimization_config(self) -> dict[str, Any]:
+        """PTQ config for MaskRCNN-Eff."""
+        if self.model_name == "maskrcnn_efficientnet_b2b":
+            return {
+                "ignored_scope": {
+                    "types": ["Add", "Divide", "Multiply", "Sigmoid"],
+                    "validate": False,
+                },
+                "preset": "mixed",
+            }
+
+        if self.model_name == "maskrcnn_swin_t":
+            return {
```

Comment:
this is updated to: `return {"model_type": "transformer"}`

Reply:
updated https://github.com/openvinotoolkit/training_extensions/pull/3865/commits/0f4409c90de69578b2e4b09e66f2b5b4c072e436